aiohue/v2/controllers/events.py

```diff
@@ -0,0 +1,191 @@
+        headers = {"Accept": "text/event-stream", "Cache-Control": "no-cache"}
+        retries = 0
+        while True:
+            if self._last_event_id:
+                headers["Last-Event-ID"] = self._last_event_id
+            try:
+                async with self._bridge.create_request(
+                    "get", "eventstream/clip/v2", timeout=0, headers=headers
+                ) as resp:
+                    # update status to connected once we reach this point
+                    self._status = EventStreamStatus.CONNECTED
+                    retries = 0  # reset on succesfull connect
+                    self._logger.debug("Connected to EventStream")
+                    # messages come in one by line, according to EventStream/SSE specs
+                    # we iterate over the incoming lines in the streamreader.
+                    # To prevent a deadlock waiting for a message while the connection is dead
+                    # we have a simple timeout guard in place.
+                    # If no message is received in 30 minutes, a Timeout error will be raised
+                    # and thus the connection re-established.
+                    iterator = resp.content.__aiter__()
+                    while True:
+                        line = await asyncio.wait_for(
+                            iterator.__anext__(), timeout=30 * 60
+                        )
+                        self.__parse_message(line)
+            except (ClientConnectionError, asyncio.TimeoutError) as err:
+                status = getattr(err, "status", None)
+                if status is not None and status == 404:
+                    raise InvalidAPIVersion from err
+                if status is not None and status == 403:
+                    raise Unauthorized from err
+                # pass all other errors because we will auto retry
+            finally:
+                self._status = EventStreamStatus.CONNECTING
+                retries += 1
+                await asyncio.sleep(2 * retries)
+
+    async def __event_processor(self) -> NoReturn:
+        """Process incoming CLIPEvents on the Queue and distribute those."""
+        while True:
+            event: CLIPEvent = await self._event_queue.get()
+            # each clip event has array of updated/added/deleted objects in data property
+            # we fire an event for each object that was added/updated/deleted
+            for item in event.data:
+                self.emit(EventType(event.type), item)
+
+    def __parse_message(self, line: bytes) -> None:
+        """Parse a plain message string as received from EventStream."""
+        try:
+            line = line.decode().strip()
+            if not line or ":" not in line:
+                return
+            key, value = line.split(":", 1)
+            if not key:
+                return
+            if key == "id":
+                self._last_event_id = value.replace(":0", "")
+            elif key == "data":
```

Comment:
Make this a guard clause.
```suggestion
                self._last_event_id = value.replace(":0", "")
                return
                
            if key != "data":
                self._logger.debug("Received unexpected message: %s - %s", key, value)

            # events is array with multiple events (but contains just one)
```